scene/main/node.h

```diff
@@ -636,7 +637,11 @@ class Node : public Object {
 
 	_FORCE_INLINE_ Viewport *get_viewport() const { return data.viewport; }
 
-	virtual PackedStringArray get_configuration_warnings() const;
+	virtual Array get_configuration_warnings() const;
+	Dictionary configuration_warning_to_dict(const Variant &p_warning) const;
+	Vector<Dictionary> get_configuration_warnings_as_dicts() const;
+	Vector<Dictionary> get_configuration_warnings_of_property(const String &p_property = String()) const;
+	PackedStringArray get_configuration_warnings_as_strings(bool p_wrap_lines, const String &p_property = String()) const;
```

Comment:
In https://github.com/godotengine/godot/pull/81892 I proposed removing the String conversion method from `Node`, as it's only used by the editor. This PR is going in the opposite direction, adding more formatting methods to `Node`.

IMO, only `virtual Array get_configuration_warnings() const;` should exist in `Node`. The rest are editor-specific.

Reply:
That is very fair. Since they are not bound methods (and shouldn't be), it's not a problem to move them elsewhere as a refactor further down the line.

I just don't know enough about organization in C++ to really know a good place for them, or if you can just create a `editor_configuration_warnings.cpp` file for this and it'll magically work?

Reply:
Note that it was also removed in https://github.com/godotengine/godot/pull/87535 for similar reasons.